core/src/test/java/org/kiwix/kiwixmobile/core/dao/NewBookmarksDaoTest.kt

```diff
@@ -0,0 +1,270 @@
+import io.objectbox.query.QueryBuilder
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.kiwix.kiwixmobile.core.CoreApp
+import org.kiwix.kiwixmobile.core.bookmark.BookmarkItem
+import org.kiwix.kiwixmobile.core.dao.entities.BookOnDiskEntity
+import org.kiwix.kiwixmobile.core.dao.entities.BookOnDiskEntity_
+import org.kiwix.kiwixmobile.core.dao.entities.BookmarkEntity
+import org.kiwix.kiwixmobile.core.dao.entities.BookmarkEntity_
+import org.kiwix.kiwixmobile.core.data.local.entity.Bookmark
+import org.kiwix.kiwixmobile.core.reader.ZimFileReader
+import org.kiwix.sharedFunctions.bookmarkItem
+import org.kiwix.sharedFunctions.bookmarkEntity
+import org.kiwix.sharedFunctions.bookOnDiskEntity
+
+internal class NewBookmarksDaoTest {
+
+  private val box: Box<BookmarkEntity> = mockk(relaxed = true)
+  private val newBookmarksDao = NewBookmarksDao(box)
+
+  @BeforeEach
+  fun init() {
+    clearAllMocks()
+    mockkStatic(CoreApp::class)
+    every { CoreApp.getInstance().packageName } returns "pkg"
+  }
+
+  @Test
+  fun `get bookmarks without current book`() {
+    val zimFileReader: ZimFileReader = mockk()
+    val queryBuilder: QueryBuilder<BookmarkEntity> = mockk()
+    every { box.query() } returns queryBuilder
+    every { queryBuilder.order(BookmarkEntity_.bookmarkTitle) } returns queryBuilder
+    val query: Query<BookmarkEntity> = mockk(relaxed = true)
+    every { queryBuilder.build() } returns query
+    val bookmarkEntities: List<BookmarkEntity> = listOf(bookmarkEntity())
+    every { query.find() } returns bookmarkEntities
+    val bookmarkItem = BookmarkItem(bookmarkEntity())
+    assertThat(newBookmarksDao.getBookmarks(false, zimFileReader))
+      .isEqualTo(listOf(bookmarkItem))
+  }
+
+  @Test
+  fun `get bookmarks from current book when zimFileReader is null`() {
+    val queryBuilder: QueryBuilder<BookmarkEntity> = mockk()
+    every { box.query() } returns queryBuilder
+    every { queryBuilder.equal(BookmarkEntity_.zimName, "") } returns queryBuilder
+    every { queryBuilder.order(BookmarkEntity_.bookmarkTitle) } returns queryBuilder
+    val query: Query<BookmarkEntity> = mockk(relaxed = true)
+    every { queryBuilder.build() } returns query
+    val bookmarkEntities: List<BookmarkEntity> = listOf(bookmarkEntity())
+    every { query.find() } returns bookmarkEntities
+    val bookmarkItem = BookmarkItem(bookmarkEntity())
+    assertThat(newBookmarksDao.getBookmarks(true, null))
+      .isEqualTo(listOf(bookmarkItem))
+  }
+
+  @Test
+  fun `get bookmarks from current book`() {
+    val zimFileReader: ZimFileReader = mockk()
+    val queryBuilder: QueryBuilder<BookmarkEntity> = mockk()
+    every { box.query() } returns queryBuilder
+    val zimName = "zimName"
+    every { zimFileReader?.name } returns zimName
+    every { queryBuilder.equal(BookmarkEntity_.zimName, zimFileReader?.name) } returns queryBuilder
```

Comment:
if you were to verify this step  becausewas called then this test would fail because

```
newBookmarksDao.getBookmarks(false, zimFileReader)
```

you passed in false